test/integration/smoke/test_vm_schedule.py

```diff
@@ -0,0 +1,177 @@
+from marvin.cloudstackAPI import (deleteVMSchedule,
+                                  createVMSchedule,
+                                  listVMSchedules,
+                                  updateVMSchedule,
+                                  enableVMSchedule,
+                                  disableVMSchedule)
+
+from marvin.lib.base import * # import all resources like Account
+from marvin.lib.common import * # import all utility methods like get_zone etc.
+
+import time
+import os
+
+class Services:
+
+    """Test VM Schedule Services
+    """
+
+    def __init__(self):
+        self.services = {
+            "domain": {
+                "name": "Domain",
+            },
+            "vm_schedule": {
+                "virtualmachineid" : 9,
+                "description": "start vm",
+                "action": "start",
+                "schedule": "30",
+                "intervaltype": "HOURLY",
+                "tag": "hello",
+                "timezone": "Asia/Kolkata"
+            }
+        }
+
+class TestVmSchedule(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        testClient = super(TestVmSchedule, cls).getClsTestClient()
+        cls.apiclient = testClient.getApiClient()
+        cls.services = testClient.getParsedTestDataConfig()
+
+        # Get attributes
+        cls.zone = get_zone(cls.apiclient, testClient.getZoneForTests())
+        cls.domain = get_domain(cls.apiclient)
+        cls.hypervisor = testClient.getHypervisorInfo()
+
+        cls.cleanup = []
+
+        template = get_suitable_test_template(
+            cls.apiclient,
+            cls.zone.id,
+            cls.services["ostype"],
+            cls.hypervisor
+        )
+
+        if template == FAILED:
+            assert False, "get_suitable_test_template() failed to return template\
+                    with description %s" % cls.services["ostype"]
+
+        cls.services["domainid"] = cls.domain.id
+        cls.services["small"]["zoneid"] = cls.zone.id
+        cls.services["templates"]["ostypeid"] = template.ostypeid
+        cls.services["zoneid"] = cls.zone.id
+
+        # Create VMs, NAT Rules etc
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            domainid=cls.domain.id
+        )
+        cls.cleanup.append(cls.account)
+
+        cls.service_offering = ServiceOffering.create(
+            cls.apiclient,
+            cls.services["service_offerings"]["tiny"]
+        )
+        cls.cleanup.append(cls.service_offering)
+
+        cls.virtual_machine = VirtualMachine.create(
+            cls.apiclient,
+            cls.services["small"],
+            templateid=template.id,
+            accountid=cls.account.name,
+            domainid=cls.account.domainid,
+            serviceofferingid=cls.service_offering.id,
+            mode=cls.zone.networktype
+        )
+   #     cls.cleanup.append(cls.virtual_machine)
```

Comment:
why code in comment? take it our or uncomment, I'd say

Reply:
Fixed